wave_matching/src/icp.cpp

```diff
@@ -102,19 +288,19 @@ void ICPMatcher::estimateLUM() {
         for (auto it = list->begin(); it != list->end(); ++it) {
             if (it->index_match > -1) {
                 corrs_aver.push_back(Eigen::Vector3f(
-                  0.5f * (this->ref->points[it->index_query].x +
-                          this->target->points[it->index_match].x),
-                  0.5f * (this->ref->points[it->index_query].y +
-                          this->target->points[it->index_match].y),
-                  0.5f * (this->ref->points[it->index_query].z +
-                          this->target->points[it->index_match].z)));
+                  0.5f * (ref->points[it->index_query].x +
```

Comment:
Any reason for why all your math is float math instead of doubles?

Reply:
yes. PCL uses floats instead of doubles

Reply:
:-1: ok